src/domain/mnemonic.ts

```diff
@@ -1,19 +1,19 @@
-import { IdentityOpts, IdentityType, Mnemonic as MnemonicTdex } from 'ldk';
+import { IdentityOpts, IdentityType, Mnemonic as MnemonicIdentity, MnemonicOpts } from 'ldk';
+import { Network } from './network';
 
 export type Mnemonic = string;
 
-export function createMnemonic(mnemo: string): Mnemonic {
+export function createMnemonic(mnemo: string, network: Network): Mnemonic {
   // Trim start-end and replace multiple spaces in between with a single space
   const mnemonic = mnemo.trim().replace(/ +(?= )/g, '');
 
   try {
     // Mnemonic validation, network doesn't matter
-    new MnemonicTdex({
-      // TODO rename
-      chain: 'liquid',
+    new MnemonicIdentity({
+      chain: network,
```

Comment:
Does the network matters now? If not we can leave it to 'liquid'

Reply:
I think we should drop the `new MnemonicIdentity` test and replace it by `bip32.validateMnemonic` function